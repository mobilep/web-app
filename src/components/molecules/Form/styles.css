.Form {
	padding: 10px;
	text-align: center;
}

.Form-textInputWrapper {
	margin-top: 10px;
}

.Form-buttonWrapper {
	margin-top: 40px;
}

.Form-header,
.Form-paragraph {
	font-size: 22px;
}

.Form-paragraph {
	margin-top: 20px;
}

.Form-form {
	margin: 0 auto;
}

@media (max-width: 768px) {
	.Form-textInputWrapper {
		margin-top: 30px;
	}

	.Form-paragraph {
		font-size: 18px;
	}
}
