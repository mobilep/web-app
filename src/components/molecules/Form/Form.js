import React, { Component, createContext } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { ButtonWithArrow } from '../../atoms';
import { TextInput, CheckboxInput } from '..';
import './styles.css';

const getClassName = (baseName, { className }) => (
	classNames(baseName, className)
);
const { Consumer, Provider } = createContext();

export default class Form extends Component {

	state = {
		showError: false,
	}

	getInputValues (formData) {
		return Array.from(formData.entries())
			.reduce((prev, curr) => {
				const [key, value] = curr;
				prev[key] = value;
				return prev;
			}, {});
	}

	handleSubmit = (event) => {
		event.preventDefault();
		this.setState({ showError: true });
		const form = event.target;
		if (!form.checkValidity()) return;
		const formData = new FormData(form);
		const inputValues = this.getInputValues(formData);
		this.props.onSubmit(inputValues);
	}

	render () {
		const { children, className, header, paragraph } = this.props;
		return (
			<div className={`Form ${className}`}>
				{
					header &&
					<div className='Form-header'>{header}</div>
				}
				{
					paragraph &&
					<div className='Form-paragraph'>{paragraph}</div>
				}

				<form
					className='Form-form'
					noValidate
					onSubmit={this.handleSubmit}
				>
					<Provider value={{ ...this.state }}>
						{children}
					</Provider>
				</form>
			</div>
		);
	}
}

Form.TextInput = (props) => (
	<Consumer>
		{({ showError }) => (
			<TextInput
				{...props} {...{ showError }}
				className={getClassName('Form-textInputWrapper', props)}
			/>)}
	</Consumer>
);

Form.Checkbox = (props) => (
	<Consumer>
		{({ showError }) => (
			<CheckboxInput
				{...props} {...{ showError }}
				className={getClassName('Form-textInputWrapper', props)}
			/>)}
	</Consumer>
);

Form.Button = (props) =>
	<ButtonWithArrow {...props} type='submit' className={getClassName('Form-buttonWrapper', props)} />;

Form.defaultProps = {
	onSubmit: () => {},
	className: '',
};

Form.propTypes = {
	children: PropTypes.node,
	className: PropTypes.string,
	header: PropTypes.string,
	onSubmit: PropTypes.func,
	paragraph: PropTypes.string,
	submitting: PropTypes.bool,
};
