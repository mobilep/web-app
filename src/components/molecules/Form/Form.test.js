import React from 'react';
import { shallow } from 'enzyme';

import Form from './Form';

const PROPS = {
	header: 'Header',
	paragraph: 'Paragraph',
};

describe('Molecule - Form', () => {

	it('Should render', () => {
		const wrapper = shallow(
			<Form {...PROPS}>
				Content
			</Form>
		);
		expect(wrapper).toMatchSnapshot();
	});

	it('Should render in submitting state', () => {
		const wrapper = shallow(
			<Form {...PROPS} submitting>
				Content
			</Form>
		);
		expect(wrapper).toMatchSnapshot();
	});

});
