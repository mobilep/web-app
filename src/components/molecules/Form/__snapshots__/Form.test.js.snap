// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecule - Form Should render 1`] = `
<div
  className="Form "
>
  <div
    className="Form-header"
  >
    Header
  </div>
  <div
    className="Form-paragraph"
  >
    Paragraph
  </div>
  <form
    className="Form-form"
    noValidate={true}
    onSubmit={[Function]}
  >
    <ContextProvider
      value={
        Object {
          "showError": false,
        }
      }
    >
      Content
    </ContextProvider>
  </form>
</div>
`;

exports[`Molecule - Form Should render in submitting state 1`] = `
<div
  className="Form "
>
  <div
    className="Form-header"
  >
    Header
  </div>
  <div
    className="Form-paragraph"
  >
    Paragraph
  </div>
  <form
    className="Form-form"
    noValidate={true}
    onSubmit={[Function]}
  >
    <ContextProvider
      value={
        Object {
          "showError": false,
        }
      }
    >
      Content
    </ContextProvider>
  </form>
</div>
`;
